### PR TITLE
Changes Capsaicin On-Touch Protection Requirements

### PR DESCRIPTION
**Pull request by Wisemonster**
[BALANCE]

## About the PR 
This changes the protection requirements of capsaicin to require both smoke cloud protection and eye wear in order to protect from it's on-touch effects.

## Why's this needed? 
Currently, it only checks if a carbon entity is wearing a mask. Considering the only entities that are susceptible to it all spawn with a box that contains a mask (I.E. everyone), it currently is useless when it comes to using the crowd dispersal grenades. This makes it so that the grenades are now a viable choice in the sec loadout vendor.

## Changelog

```changelog
(u)Wisemonster
(*)Capsaicin (the stuff in crowd dispersal grenades and peppers) now requires chem smoke protection (or internals) and glasses to be protected from it.
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 1]
```diff
@@ -2043,8 +2043,11 @@ datum
 
 
 				else if (method == TOUCH)
-					if(iscarbon(M))
-						if(!M.wear_mask)
+					if(iscarbon(M) || ismobcritter(M))
+						var/mob/living/carbon/human/H = M
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:2047]: Mob critters aren't `mob/living/carbon/human`.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 2]
```diff
@@ -2043,8 +2043,11 @@ datum
...
+						if((issmokeimmune(H) && H.glasses?.c_flags & COVERSEYES) || isrobocritter(M) || /mob/living/critter/fire_elemental) // robotic critters and fire elementals will be immune, but not organic critters.
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:2048]: Missing an `istype`.

**Review by flrsh — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Reagents-FoodDrink.dm (hunk 1 above)
Comment (Wisemonster) [reply code/modules/chemistry/Reagents-FoodDrink.dm:2047]: Aight, added a var for critters.

File: code/modules/chemistry/Reagents-FoodDrink.dm (hunk 2 above)
Comment (Wisemonster) [reply code/modules/chemistry/Reagents-FoodDrink.dm:2048]: Fixed

**Pull request merged**